vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ImageServiceTest.java

```diff
@@ -61,4 +90,76 @@ public void test_getImageDimensions_fullHeight() {
         Assert.assertEquals(333, data.getWidth());
         Assert.assertEquals(200, data.getHeight());
     }
+    
+    @Test
+    public void test_getTotalPages_success() {
+        when(imageRepo.count()).thenReturn(12L);
+        assertEquals(2, serviceToTest.getTotalPages());
+    }
+    
+    @Test
+    public void test_getTotalPages_whenZeroImages() {
+        when(imageRepo.count()).thenReturn(0L);
+        assertEquals(0, serviceToTest.getTotalPages());
+    }
+   
+    @Test
+    public void test_getImages_success() { 
+        Pageable sortByRequestedField = PageRequest.of(0, 10, Sort.by(SortByField.CREATION_DATE.getValue()));
+        when(imageRepo.count()).thenReturn(1L);
+        when(imageRepo.findAll(sortByRequestedField)).thenReturn(new PageImpl<VSImage>(images));
+        List<VSImage> requestedImages = serviceToTest.getImages(1);
+        assertEquals(IMG_ID, requestedImages.get(0).getId());
+        verify(imageRepo).findAll(sortByRequestedField);
+    }
+  
+    @Test
+    public void test_getImages_negativePage() { 
+        Pageable sortByRequestedField = PageRequest.of(0, 10, Sort.by(SortByField.CREATION_DATE.getValue()));
+        when(imageRepo.count()).thenReturn(1L);
+        when(imageRepo.findAll(sortByRequestedField)).thenReturn(new PageImpl<VSImage>(images));
+        List<VSImage> requestedImages = serviceToTest.getImages(-2);
+        assertEquals(IMG_ID, requestedImages.get(0).getId());
+        verify(imageRepo).findAll(sortByRequestedField);
+    }
+    
+    @Test
+    public void test_getImages_pageGreaterThanTotalPages() { 
+        Pageable sortByRequestedField = PageRequest.of(0, 10, Sort.by(SortByField.CREATION_DATE.getValue()));
+        when(imageRepo.count()).thenReturn(1L);
+        when(imageRepo.findAll(sortByRequestedField)).thenReturn(new PageImpl<VSImage>(images));
+        List<VSImage> requestedImages = serviceToTest.getImages(5);
+        assertEquals(IMG_ID, requestedImages.get(0).getId());
+        verify(imageRepo).findAll(sortByRequestedField);
```

Comment:
to really test this you need more than one page. there could be a bug in which, the first page would requested when the given page number is greater than the total. You wouldn't catch this with this test.